core/src/main/java/tech/pegasys/eth2signer/core/Eth2Signer.java

```diff
@@ -0,0 +1,110 @@
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core;
+
+import tech.pegasys.eth2signer.core.http.LogErrorHandler;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.handler.ResponseContentTypeHandler;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Eth2Signer implements Runnable {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final String TEXT = HttpHeaderValues.TEXT_PLAIN.toString() + "; charset=utf-8";
+  private final Config config;
+
+  public Eth2Signer(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run() {
+    final HttpServerOptions serverOptions =
+        new HttpServerOptions()
+            .setPort(config.getHttpListenPort())
+            .setHost(config.getHttpListenHost())
+            .setReuseAddress(true)
+            .setReusePort(true);
+
+    final Vertx vertx = Vertx.vertx();
+    try {
+      final Router router = Router.router(vertx);
+      router
+          .route(HttpMethod.GET, "/upcheck")
+          .produces(TEXT)
+          .handler(BodyHandler.create())
+          .handler(ResponseContentTypeHandler.create())
+          .failureHandler(new LogErrorHandler())
+          .handler(routingContext -> routingContext.response().end("OK"));
+
+      final HttpServer httpServer = vertx.createHttpServer(serverOptions);
+      final CompletableFuture<Void> serverRunningFuture = new CompletableFuture<>();
+      httpServer
+          .requestHandler(router)
+          .listen(
+              result -> {
+                if (result.succeeded()) {
+                  serverRunningFuture.complete(null);
+                } else {
+                  serverRunningFuture.completeExceptionally(result.cause());
+                }
+              });
+      serverRunningFuture.get();
+      LOG.info("Server is up, and listening on {}", httpServer.actualPort());
+      if (config.getDataPath() != null) {
+        final File portsFile = new File(config.getDataPath().toFile(), "eth2signer.ports");
+        portsFile.deleteOnExit();
+        writePortsToFile(portsFile, httpServer.actualPort());
+      }
+    } catch (final ExecutionException | InterruptedException e) {
+      vertx.close();
+      throw new RuntimeException("Failed to create Http Server", e.getCause());
+    } catch (final Throwable t) {
+      vertx.close();
```

Comment:
could simplify the catches by putting the vertx.close in a finally and not catching throwable and throwing

Reply:
putting in a finally would mean vertx closes regardless of exception - so we occur immediately.